stravalib/model.py

```diff
@@ -116,6 +125,80 @@ def __repr__(self):
         )
 
 
+class DeprecatedSerializableMixin(BaseModel):
+    """
+    Provides backward compatibility with legacy BaseEntity
+    """
+
+    @classmethod
+    def deserialize(cls, v):
+        """
+        Creates and returns a new object based on serialized (dict) struct.
+        """
+        warn_method_deprecation(
+            cls,
+            "deserialize()",
+            "parse_obj()",
+            "https://docs.pydantic.dev/usage/models/#helper-functions",
+        )
+        return cls.parse_obj(v)
+
+    def from_dict(self, v):
```

Comment:
sorry is this deprecated? where is from_dict being defined if it is deprecated here?

This workf locally but i find the variable name `v` to be confusing. I suspect v needs to be a `dict` with the club attributes - 

and is the intended workflow

to_dict --> update a few fields and then 
from_dict - > update the club metadata ?

then maybe push back to strava to update?

So sorry if these are basic questions but i'm trying to understand how this is used and hoping we can clarify what these methods do via better variable names, maybe some type hinting? and better docstrings (i can help too where needed with that part). you are the pydantic model guru here so i'm trying to better understand this shift while also thinking about usability and use cases.

Reply:
These (de-)serialization methods are mainly used internally by other (legacy) `model` and `client` code, but are defined as public methods. Simply discarding them may break existing functionality in projects dependent on stravalib.

The preference (which I think we should document) is to use pydantic's (de-)serialization API, which would look like:

```python
c = Club.parse_obj(some_club_dict)
club_as_dict = c.dict() # ... and push to document store
club_as_json = c.json() # ... and send to other API
```

See https://docs.pydantic.dev/usage/models/#helper-functions and https://docs.pydantic.dev/usage/exporting_models/

Reply:
Oh i get it ... and that part of the API will be more expressive . 

so essentially the use case that i'm hearing here is this is for someone building an app, that might want to grab say club data and add it to a database or wherever they are storing app data via another API. (which might be what you're doing with the application that you're working on?)

This makes a lot of sense and the pydantic docs are really helpful. i'll spend some more time on them. 
So as far as i can tell, this works for me locally and seems seamless.

Reply:
Yes, that's the use case I thought about (and often use myself). Something similar is getting activities from Strava and transforming them into a Pandas DataFrame, which would look like
```python
activity_df = pd.DataFrame([a.dict() for a in client.get_activities()])
```
once the `Activity` class is transformed.